src/services/twilio-api/twilio-client.js

```diff
@@ -180,12 +181,12 @@ class TwilioApiClient {
       opts.headers.Accept = 'application/json';
     }
 
-    if (!opts.uri) {
-      if (opts.path.includes(TwilioApiFlags.ACCOUNT_SID) && !doesObjectHaveProperty(opts.pathParams, TwilioApiFlags.ACCOUNT_SID)) {
-        opts.pathParams[TwilioApiFlags.ACCOUNT_SID] = this.accountSid;
-      }
+    if (opts.path.includes(TwilioApiFlags.ACCOUNT_SID) && !doesObjectHaveProperty(opts.pathParams, TwilioApiFlags.ACCOUNT_SID)) {
+      opts.pathParams[TwilioApiFlags.ACCOUNT_SID] = this.accountSid;
```

Comment:
Why was this changed?

Reply:
path params weren't being substituted if there was a uri. For example, in the tests where `opts.uri` was defined and a path representing the endpoint was provided via `opts.path`, it would not append the path to the url.

Reply:
This only applied to the account SID path param.

Reply:
The endpoint used for the tests did have account sid as the only path param, so I guess it wouldn't be an issue for other endpoints. Why wouldn't we want this substitution made if `opts.uri` is provided?

Reply:
I think the reason the check was there originally is because only the `list` operation would ever supply a `uri` (when getting the next page of results). Other operations would not include a `uri` because it would be built downstream. The next page `uri` should not be parameterized and thus would not need any `pathParams` to be set. In fact, `pathParams` is not a key in `opts` in this case (see `list` implementation above) so attempting to update path params would cause an exception and so a check was added for the uri.

Reply:
So I guess the question is, if we never expect the `opts.uri` to contain a region/edge, should we even account for these situations? Should those scenarios even be tested? The only reason I removed that check was to account for situations were `opts.uri` was of the form `https://api.edge.region.twilio.com`

Reply:
`opts.uri` may or may not contain edge/region. But it should never be parameterized.

Reply:
Wouldn't this break the `nextPageUri` test, then? (note: it did break because of this change, but was fixed with the additional path check in `getUri`

Reply:
Not sure which test that is, but `pathParams` will be undefined in the subsequent calls to `request` from `list`. There are no path params when paging because the `uri` is expected to be fully-formed already.

Reply:
`opts.pathParams` defaults to `opts.pathParams || {}`. Wouldn't that mean in subsequent calls, `opts.pathParams === {}`, which would just fail the condition in the if?

Reply:
Ah, I missed that line. But either way, the original intent was to illustrate that when a `uri` is given, `pathParams` don't do anything.

Reply:
Ah, understood 👍